integration/read_write_mode_test.go

```diff
@@ -70,3 +75,30 @@ func TestReadWriteMode(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, []string{"__name__", "foo"}, labelNames)
 }
+
+func expectedMatrix(series []prompb.TimeSeries) model.Matrix {
+	m := make(model.Matrix, 0, len(series))
+
+	for _, s := range series {
+
+		m = append(m, &model.SampleStream{
+			Metric: prompbLabelsToModelMetric(s.Labels),
+			Values: prompbSamplesToModelSamplePairs(s.Samples),
+		})
+	}
+
+	return m
+}
+
+func prompbSamplesToModelSamplePairs(samples []prompb.Sample) []model.SamplePair {
+	v := make([]model.SamplePair, 0, len(samples))
+
+	for _, s := range samples {
+		v = append(v, model.SamplePair{
+			Timestamp: model.TimeFromUnixNano(s.Timestamp * 1000000),
```

Comment:
`s.Timestamp`  is already millis, right? So we can just cast it.

```suggestion
			Timestamp: model.Time(s.Timestamp),
```

Reply:
Good catch, thanks.